hazelcast/src/main/java/com/hazelcast/query/impl/TypeConverters.java

```diff
@@ -37,20 +37,26 @@
     public static final TypeConverter SQL_DATE_CONVERTER = new SqlDateConverter();
     public static final TypeConverter SQL_TIMESTAMP_CONVERTER = new SqlTimestampConverter();
     public static final TypeConverter DATE_CONVERTER = new DateConverter();
+    public static final TypeConverter IDENTITY_CONVERTER = new IdentityConverter();
+    public static final TypeConverter UUID_CONVERTER = new UUIDConverter();
 
     private TypeConverters() {
     }
 
-    public interface TypeConverter {
-        Comparable convert(Comparable value);
-    }
+    public abstract static class TypeConverter {
+        abstract Comparable convertInternal(Comparable value);
 
-    static class EnumConverter implements TypeConverter {
-        @Override
         public Comparable convert(Comparable value) {
```

Comment:
It'd be good to mark this method as `final` since its not intended to be overriden